### ISSUE TITLE
Question about custom training

### ISSUE THREAD

**Issue by jasperblee** [spreka/biomagdsb#12]
Hello,

We have been using the nucleAIzer approach for the last few months to assist in the segmentation of tumor samples. At this point, we wish to perform slight modifications to the network in order to optimize it for our data – primarily due to the presence of dense, blurry nuclei clusters in the sample. We are new to the field of machine learning, and as such were wondering if you could provide some brief guidance on how best to proceed? Specifically – 

1)	Would image style transfer, or completely new training, be the best way to proceed here? For both, how much annotated data would generally be required in order to observe a meaningful increase in performance?
2)	Is there some way to manually provide feedback to the network – say, to identify nuclei which are incorrectly segmented – and use this to modify the network so that it is optimized for our situation?

Thank you for developing such a useful tool!

**Comment by spreka** [comment]
@jasperblee Thank you for your keen interest in using our approach :)
I suppose this is related to [your previous issue](https://github.com/spreka/biomagdsb/issues/10#issue-681544362).

If you wish to improve segmentation performance, I would suggest to try image style transfer training indeed. In case you are working with images of the same experiment - given they look similar, like the example from above - you can take a look at `start_singleExperimentPreparation.sh` (or .bat depending on your system), feeding your input test images (without annotations) first, by default it uses the presegmented masks returned by e.g. `start_prediction_fast.sh` or you can specify which mask folder to use [here](https://github.com/spreka/biomagdsb/blob/e6939e2b72e3dd7f52937aebf21a1797cc87e122/run_workflow_prepareStyle4singleExperiment.sh#L20).
You can start training with the synthetic images generated here and see how the new model performs. Make sure to point the training script to the new model in [`/kaggle_workflow/maskrcnn/config/train/train.json`](https://github.com/spreka/biomagdsb/blob/e6939e2b72e3dd7f52937aebf21a1797cc87e122/kaggle_workflow/maskrcnn/config/train/train.json#L7).

Alternatively, you can also annotate a couple of images to have more training data; you can also use those annotated masks as input to style transfer (instead of presegmented masks for better performance). I have an ImageJ/Fiji plugin [AnnotatorJ](https://github.com/spreka/annotatorj) which can export masks as expected by this pipeline (and some convenient functions too).

In its current form, such manual feedback cannot be given to the network; however, you can utilize the segmentation results manually corrected as "semi-automatic" input (training data). Bear in mind though, that you can introduce bias with this, so make sure to check the automatically generated masks before training on them. You can quickly check and edit masks in `batch mask mode` in AnnotatorJ, see [the docs](https://github.com/spreka/annotatorj/blob/master/AnnotatorJ_documentation.pdf).

Additionally, if your images do not look so similar (e.g. on some of them there are clumped regions, others are sparsely covered in cells, some are out of focus, some are overexposed etc.), use automatic clustering instead of the single experiment I suggested above. This is the default way to process the input data when you run the training script `start_training.sh`.

**Comment by jasperblee** [comment]
Great, thank you for the advice!
